docs/deployment/configuration/resource_manager.rst

```diff
@@ -0,0 +1,113 @@
+
+The below attribute is configurable within FlytePropeller, which can be disabled with:
+
+.. code-block:: yaml
+
+    resourcemanager:
+      type: noop
+
+The ResourceManager provides a task-type-specific pooling system for Flyte tasks. Optionally, plugin writers can request resource allocation in their tasks.
+
+A plugin defines a collection of resource pools using its configuration. Flyte uses tokens as a placeholder to represent a unit of resource.
+
+How does a Flyte plugin request for resources?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The Flyte plugin registers the resource and the desired quota of every resource with the **ResourceRegistrar** when setting up FlytePropeller. When a plugin is invoked, FlytePropeller provides a proxy for the plugin. This proxy facilitates the plugin's view of the resource pool by controlling operations to allocate and deallocate resources.
+
+.. dropdown:: :fa:`info-circle` Enabling Redis instance
+   :title: text-muted
+   :animate: fade-in-slide-down
+
+   The ResourceManager can use a Redis instance as an external store to track and manage resource pool allocation. By default, it is disabled, and can be enabled with:
+
+   .. code-block:: yaml
+
+       resourcemanager:
+          type: redis
+          resourceMaxQuota: 100
+          redis:
+            hostPaths:
+              - foo
+            hostKey: bar
+            maxRetries: 0
+
+Once the setup is complete, FlytePropeller builds a ResourceManager based on the previously requested resource registration. Based on the plugin implementation's logic, resources are allocated and deallocated.
+
+During runtime, the ResourceManager:
+
+#. Allocates tokens to the plugin.
+#. Releases tokens once the task is completed.
+
+How are resources allocated?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When a Flyte task execution needs to send a request to an external service, the plugin claims a unit of the corresponding resource. This is done using a **ResourceName**, which is a unique token and a fully qualified resource request (which is typically an integer). The execution generates this unique token and registers this token with the ResourceManager by calling the ResourceManager’s **"AllocateResource function"**. If the resource pool has sufficient capacity to fulfil your request, then the resources requested are allocated, and the plugin proceeds further.
+
+When the status is **"AllocationGranted"**, the execution moves forward and sends out the request for those resources.
```

Comment:
```suggestion
When the status changes to **"AllocationGranted"**, the execution sends out the request for those resources.
```

Reply:
I'm not clear on whose status this is....is it the status of the ResourceManager?

Reply:
the status of the allocation for the unit of the resource I believe